crates/accelerate/src/nlayout.rs

```diff
@@ -27,16 +30,28 @@ macro_rules! qubit_newtype {
         #[derive(
             Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, IntoPyObject, IntoPyObjectRef,
         )]
-        pub struct $id(pub u32);
+        #[repr(transparent)]
+        pub struct $id(pub Qubit);
 
         impl $id {
             #[inline]
-            pub fn new(val: u32) -> Self {
-                Self(val)
+            pub fn new(val: usize) -> Self {
```

Comment:
If we want to encapsulate the storage of bit types in the future, we might want to change this to `val: Qubit` now.

Reply:
There's no plan to have `new` take a `Qubit` - there's no need. Since `Qubit` now manages the validity of the object, it's totally safe to have the tuple constructor just be public, since you can't make an invalid `PhysicalQubit` by using it. `new` then matches `Qubit::new`, which is `usize` in the same way that `NodeIndex` does that - a technically fallible conversion, but the panicking constructor for it.